common/node/nns_ros_bridge.cc

```diff
@@ -0,0 +1,172 @@
+       * unsigned int,conversion from size_t to unsigned int is requried.
+       */
+      guint each_dim_stride = 1;
+
+      for (guint k = 0; k <= NNS_TENSOR_RANK_LIMIT - (j + 1); ++k) {
+        each_dim_stride *= tensors_info->info[i].dimension[k];
+      }
+      each_dim.stride = each_dim_stride;
+      each_dim.label = tensor_element_typename[tensors_info->info[i].type];
+      each_dim.size =
+          tensors_info->info[i].dimension[NNS_TENSOR_RANK_LIMIT - (j + 1)];
+
+      this->topic_layouts_pub[i].dim.push_back (each_dim);
+    }
+
+    g_return_val_if_fail (
+        this->topic_layouts_pub[i].dim.at(0).stride == tensor_size, FALSE
+        );
+  }
+
+  this->ready_to_pub = TRUE;
+  this->nns_tensors_info_pub = (GstTensorsInfo *) &(conf->info);
+
+  return ready_to_pub;
+}
+
+/**
+ * @brief	A method for publishing a ROS topic that contains the tensors passed by the NNStreamer framework
+ * @param[in] num_tensors : The number of tensors included in tensors_mem (for the verification purpose)
+ * @param[in] tensors_mem : The pointer of containers consists of information and raw data of tensors
+ * @return TRUE if the configuration is successfully done
+ */
+gboolean NnsRosBridge::publish (const guint num_tensors,
+    const GstTensorMemory *tensors_mem)
+{
+  nns_ros_bridge::tensors tensors_msg;
+
+  g_return_val_if_fail (ros::ok(), FALSE);
+  g_return_val_if_fail (this->ready_to_pub, FALSE);
+  g_return_val_if_fail (num_tensors == this->num_of_tensors_pub, FALSE);
+
+  for (guint i = 0; i < this->num_of_tensors_pub; ++i) {
+    std_msgs::UInt8MultiArray each_tensor;
+    uint8_t *src_data = (uint8_t *) tensors_mem[i].data;
+
+    each_tensor.layout = this->topic_layouts_pub[i];
+    each_tensor.data.assign (src_data, src_data + tensors_mem[i].size);
+
+    tensors_msg.tensors.push_back (each_tensor);
+  }
+
+  this->ros_sink_pub.publish (tensors_msg);
+  ros::spinOnce();
+
+  return TRUE;
+}
+
+/**
+ * C functions for NNStreamer plugins that want to publish or subscribe
+ * ROS topics. nns_ros_bridge_init() should be invoked before configuring the
+ * tensors information to publish or subscribe. Each NNStreamer plugin which
+ * uses this class should be holding the instance of NnsRosBridge (i.e., the
+ * void type pointer what the nns_ros_bridge_init() function returns).
+ */
+extern void *
+nns_ros_bridge_init (const char *node_name, const char *topic_name)
+{
+  return new NnsRosBridge (node_name, topic_name);
+}
+
+extern gboolean
+nns_ros_bridge_publish (void *instance, const guint num_tensors,
+    const GstTensorMemory *tensors_mem)
+{
+  NnsRosBridge *nrb_instance = (NnsRosBridge *) instance;
+
+  return nrb_instance->publish (num_tensors, tensors_mem);
+}
+
+extern gboolean
+nns_ros_bridge_set_pub_topic (void *instance, const GstTensorsConfig *conf)
+{
+  NnsRosBridge *nrb_instance = (NnsRosBridge *) instance;
+
+  return nrb_instance->setPubTopicInfo (conf);
+}
```

Comment:
Check your editor.

Reply:
@myungjoo  Fixed and **"files.insertFinalNewline": true** is  set to the global setting of my VS code now.